### PR TITLE
Num params fix

### PR DESCRIPTION
**Pull request by cvjjm**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [x] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

Please see dicussion [over here](https://github.com/PennyLaneAI/pennylane/issues/2099)

**Description of the Change:**

Makes num_params a static class property.

**Benefits:**

Makes num_params accessible to users again that do not have an instance of an Operation class.

**Possible Drawbacks:**

None.

**Related GitHub Issues:**

#2099

### CODE REVIEW

File: pennylane/ops/qubit/non_parametric_ops.py [hunk 1]
```diff
@@ -777,8 +738,8 @@ class CSWAP(Operation):
     Args:
         wires (Sequence[int]): the wires the operation acts on
     """
-    is_self_inverse = True
```
Comment (mariaschuld) [inline pennylane/ops/qubit/non_parametric_ops.py:780]: Whoops what happened here? Seems to be unrelated to this PR? [This line should be deleted on the op-refactor branch anyways, but cleaner to leave it for now?]

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 <h3>New features since last release</h3>
 
+* The num_params property of Operations is again a static property of the class.
+  This allows to programatically know the number of parameters before an Operation
+  is instanciated, which is only possible if the number of parameters is known.
```
Comment (mariaschuld) [inline doc/releases/changelog-dev.md:9]: Remember adding the PR number and link.

I'd suggest being more specific:

```suggestion
* For subclasses of `Operator` where it is known before instantiation, the `num_params` is reverted back to being a 
   static property. This allows to programmatically know the number of parameters before an operator is 
   instantiated, and does not change the user interface.
```

**Review by mariaschuld — CHANGES_REQUESTED** [review]
Thanks for making the PR. 

Since the UI does not change at all with the added benefit you aim for, this should be really nice. We should explain this logic (i.e. some attributes are static some are not) but I'll take care of this on the op-refactor branch.

Just two minor comments, and I think a merge conflict has come up.

File: pennylane/ops/qubit/non_parametric_ops.py (hunk 1 above)
Comment (cvjjm) [reply pennylane/ops/qubit/non_parametric_ops.py:780]: Oh! Good catch! I started this PR, noticed and deleted this line, then had a meeting, wanted to put it back in and notify you but forgot, submitted, and did not check everything in the end. I think this should be removed (and you seem to agree), but I agree this is not conceptually part of this PR. I will revert the change. Can you take care of implementing this in the op-refactor branch?

**Comment by cvjjm** [comment]
Thanks for reviewing @rmoyard. Now everything should be in order.

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 <h3>New features since last release</h3>
 
+* For subclasses of `Operator` where it is known before instantiation, the `num_params` is reverted back to being a 
+   static property. This allows to programmatically know the number of parameters before an operator is 
+   instantiated without changing the user interface.
+   [(#2099)](https://github.com/PennyLaneAI/pennylane/issues/2099)
```
Comment (mariaschuld) [inline doc/releases/changelog-dev.md:10]:
```suggestion
* For subclasses of `Operator` where it is known before instantiation, the `num_params` is reverted back to being a 
  static property. This allows to programmatically know the number of parameters before an operator is 
  instantiated without changing the user interface.
  [(#2099)](https://github.com/PennyLaneAI/pennylane/issues/2099)
```

**Review by mariaschuld — APPROVED** [review]

**Comment by mariaschuld** [comment]
Codecov complains, but shows that "no lines changed coverage". 

Indeed, if the property has been tested, the static attribute should be now tested, so there should not be a difference in coverage.

I will just overwrite!
